### PR TITLE
feat(generate-event): Add Http API V2 event

### LINKED ISSUE #3654: generate-event: Add API Gateway Http API V2
### Describe your idea/feature/enhancement

API Gateway V2 (Http API) payload format V2 is missing from `sam local generate-event`

### Proposal

Add API Gateway V2 - Http API payload format V2 event to generate-event

### Additional Details

- [Working with AWS Lambda proxy integrations for HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html)

Payload format version V2 Example

```json5
{
  "version": "2.0",
  "routeKey": "$default",
  "rawPath": "/my/path",
  "rawQueryString": "parameter1=value1&parameter1=value2&parameter2=value",
  "cookies": [
    "cookie1",
    "cookie2"
  ],
  "headers": {
    "header1": "value1",
    "header2": "value1,value2"
  },
  "queryStringParameters": {
    "parameter1": "value1,value2",
    "parameter2": "value"
  },
  "requestContext": {
    "accountId": "123456789012",
    "apiId": "api-id",
    "authentication": {
      "clientCert": {
        "clientCertPem": "CERT_CONTENT",
        "subjectDN": "www.example.com",
        "issuerDN": "Example issuer",
        "serialNumber": "a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1",
        "validity": {
          "notBefore": "May 28 12:30:02 2019 GMT",
          "notAfter": "Aug  5 09:36:04 2021 GMT"
        }
      }
    },
    "authorizer": {
      "jwt": {
        "claims": {
          "claim1": "value1",
          "claim2": "value2"
        },
        "scopes": [
          "scope1",
          "scope2"
        ]
      }
    },
    "domainName": "id.execute-api.us-east-1.amazonaws.com",
    "domainPrefix": "id",
    "http": {
      "method": "POST",
      "path": "/my/path",
      "protocol": "HTTP/1.1",
      "sourceIp": "IP",
      "userAgent": "agent"
    },
    "requestId": "id",
    "routeKey": "$default",
    "stage": "$default",
    "time": "12/Mar/2020:19:03:58 +0000",
    "timeEpoch": 1583348638390
  },
  "body": "Hello from Lambda",
  "pathParameters": {
    "parameter1": "value1"
  },
  "isBase64Encoded": false,
  "stageVariables": {
    "stageVariable1": "value1",
    "stageVariable2": "value2"
  }
}
```

- [apigw-v2-request-iam.json](https://github.com/aws/aws-lambda-go/blob/main/events/testdata/apigw-v2-request-iam.json)

```json
{
    "version": "2.0",
    "routeKey": "$default",
    "rawPath": "/my/path",
    "rawQueryString": "parameter1=value1&parameter1=value2&parameter2=value",
    "cookies": [
        "cookie1",
        "cookie2"
    ],
    "headers": {
        "Header1": "value1",
        "Header2": "value2"
    },
    "queryStringParameters": {
        "parameter1": "value1,value2",
        "parameter2": "value"
    },
    "pathParameters": {
        "proxy": "hello/world"
    },
    "requestContext": {
        "routeKey": "$default",
        "accountId": "123456789012",
        "stage": "$default",
        "requestId": "id",
        "authorizer": {
            "iam": {
                "accessKey": "ARIA2ZJZYVUEREEIHAKY",
                "accountId": "1234567890",
                "callerId": "AROA7ZJZYVRE7C3DUXHH6:CognitoIdentityCredentials",
                "cognitoIdentity": {
                    "amr" : ["foo"],
                    "identityId": "us-east-1:3f291106-8703-466b-8f2b-3ecee1ca56ce",
                    "identityPoolId": "us-east-1:4f291106-8703-466b-8f2b-3ecee1ca56ce"
                },
                "principalOrgId": "AwsOrgId",
                "userArn": "arn:aws:iam::1234567890:user/Admin",
                "userId": "AROA2ZJZYVRE7Y3TUXHH6"
            }
        },
        "apiId": "api-id",
        "authentication": {
            "clientCert": {
                "clientCertPem": "-----BEGIN CERTIFICATE-----\nMIIEZTCCAk0CAQEwDQ...",
                "issuerDN": "C=US,ST=Washington,L=Seattle,O=Amazon Web Services,OU=Security,CN=My Private CA",
                "serialNumber": "1",
                "subjectDN": "C=US,ST=Washington,L=Seattle,O=Amazon Web Services,OU=Security,CN=My Client",
                "validity": {
                    "notAfter": "Aug  5 00:28:21 2120 GMT",
                    "notBefore": "Aug 29 00:28:21 2020 GMT"
                }
            }            
        },
        "domainName": "id.execute-api.us-east-1.amazonaws.com",
        "domainPrefix": "id",
        "time": "12/Mar/2020:19:03:58+0000",
        "timeEpoch": 1583348638390,
        "http": {
            "method": "GET",
            "path": "/my/path",
            "protocol": "HTTP/1.1",
            "sourceIp": "IP",
            "userAgent": "agent"
        }
    },
    "stageVariables": {
        "stageVariable1": "value1",
        "stageVariable2": "value2"
    },
    "body": "{\r\n\t\"a\": 1\r\n}",
    "isBase64Encoded": false
}
```

### PR DESCRIPTION
**Pull request by michaelbrewer**
#### Which issue(s) does this change fix?

- #3654

#### Why is this change necessary?

When working with API Gateway HTTP API V2 Payload, it would be nice to have a way to generate sample events

#### How does it address the issue?

Add HTTP API V2 payload example

#### What side effects does this change have?

- Docs will also need to be updated: [docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-local-generate-event.html](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-local-generate-event.html)

#### Checklist

- [X] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [X] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [X] Write unit tests
- [X] Write/update functional tests
- [X] Write/update integration tests
- [X] `make pr` passes
- [X] `make update-reproducible-reqs` if dependencies were changed
- [X] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/lib/generated_sample_events/event-mapping.json [hunk 1]
```diff
@@ -190,7 +190,35 @@
...
+          "default": "POST"
+        },
+        "path": {
+          "type": "string",
+          "default": "path/to/resource"
+        },
+        "account-id": {
+          "default": "123456789012"
+        },
+        "dns-suffix": {
```
Comment (moelasmar) [inline samcli/lib/generated_sample_events/event-mapping.json:218]: this parameter is not used.

**Review by moelasmar — CHANGES_REQUESTED** [review]

File: samcli/lib/generated_sample_events/event-mapping.json (hunk 1 above)
Comment (michaelbrewer) [reply samcli/lib/generated_sample_events/event-mapping.json:218]: dns-suffix is used as `dns_suffix` due to how the templating works.:

```json
    "domainName": "id.execute-api.{{{dns_suffix}}}",
```

Comment (moelasmar) [reply samcli/lib/generated_sample_events/event-mapping.json:218]: sorry it is my bad

**Review by moelasmar — APPROVED** [review]

**Review by mingkun2020 — APPROVED** [review]

**Comment by torresxb1** [comment]
Thank you for the contribution @michaelbrewer!

**Pull request merged**